### PR TITLE
Api 9801 lighthouse token validator add rewrite rule

### PR DESCRIPTION
**Pull request by lukemarkwordtlibertyits**
# Description

The lighthouse-token-validator now accepts the following rewrite rule `icn and default`. If no rule is specified in the config default will be applied.

## Rules

**Default:** The path is not changed.

**Icn:** An icn template is appended to the path.

## Quick Start

See readme for config.

## Tests

**Unit:** `./run-unit-tests.sh`

**Regression**: see `tests/lighthouse-token-validator/README.md`

## Related PRs

// TODO k8 config ( this is not a blocking PR )

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,7 @@ api_key - Required if the token_verification_mode is oauth_verification. Determi
 audience - Required if the token_verification_mode is oauth_verification. Determines the 'aud value to be checked at the oauth validation endpoint.
 scope_check - whether or not the token will be checked for a particular scope.
 scopes - Defines a list of scopes to check for. ex `scopes = [ "patient/Observation.read", "offline_access" ]`
+rewrite_rule - Defines how the request url is re-writen. Options {icn or default}
```
Comment (bschofield-va) [inline README.md:62]: I think you need to explain what these modes actually do. As a user of this plugin, I do not know what 'icn' rewrite_rule does and when I might want or need it.

**Comment by bschofield-va** [comment]
What's the status with this PR?

**Comment by dbrown-va** [comment]
> What's the status with this PR?

Actively reviewing.  Running into some strange dns mapping failures while exercising locally with docker-compose.

Hope to get it finished up today/tomorrow.

**Comment by dbrown-va** [comment]
Running into DNS errors when attempting to call dev-api token validation.

Not isolated to the new plugin, set up health-apis-token-validator in a local docker-compose and ran into the same problem.

Temporarily pointing to sandbox or staging resolves the issue.  Devops has been notified and is discussing.

**Review by dbrown-va — APPROVED** [review]
After pointing at sandbox for the token validation call confirmed this functionality for both the ICN and default rewrites, as well as when the config is not preset.

File: kong/plugins/lighthouse-token-validator/path_rewrite_factory.lua [hunk 2]
```diff
@@ -0,0 +1,68 @@
+local cjson = require "cjson.safe"
+
+function FuncNew( obj ) -- for Inheritance
```
Comment (dbrown-va) [inline kong/plugins/lighthouse-token-validator/path_rewrite_factory.lua:3]: I would not be opposed to a summary comment here that explained in a sentence or two how an additional rewrite run would be added.  Only one I can think off of the top of my head is something like a sec_id vs an icn.

**Review by bschofield-va — APPROVED** [review]

**Review by esubocz-va — APPROVED** [review]
Also able to confirm functionality with sandbox validation. All regression tests pass. LGTM.

**Pull request merged**